gslib/tests/test_rsync.py

```diff
@@ -3291,11 +3421,13 @@ def test_bucket_to_bucket_includes_arbitrary_headers(self):
         suri(bucket2_uri)
     ],
                             return_stderr=True)
-
-    headers_for_all_requests = re.findall(r"Headers: \{([\s\S]*?)\}", stderr)
-    self.assertTrue(headers_for_all_requests)
-    for headers in headers_for_all_requests:
-      self.assertIn("'arbitrary': 'header'", headers)
+    if self._use_gcloud_storage:
+      self.assertIn('--additional-headers=arbitrary=header', stderr)
```

Comment:
I think this test was intended to ensure that all requests contained the additional header, so this assertion is a bit weaker than the gsutil one. Is it possible to do a similar check here?

Reply:
Good call. Done.